Detections/MultipleDataSources/UserImpersonateByAAID.yaml

```diff
@@ -0,0 +1,67 @@
+id: 11c3d541-5fa5-49df-8218-d1c98584473b
+name: User impersonation by Identity Protection alerts
+description: |
+  'This detection focuses on identifying user-related events involving IAM roles, groups, user access, and password changes. It examines instances where the user's IP address matches and alerts generated by Identity Protection share the same IP address. The analysis occurs within a time window of 1 hour, helping to flag potential cases of user impersonation.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureActiveDirectoryIdentityProtection
+    dataTypes:
+      - SecurityAlert (IPC)
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
+query: |
+ // Retrieve SecurityAlerts generated within the last day
+  SecurityAlert 
+  | where TimeGenerated >= ago(1d)
+  // Filter alerts for Azure Active Directory Identity Protection and High severity
+  | where ProductName has "Azure Active Directory Identity Protection"

```

Comment:
Why are we just limiting to IDP here and not other products as well?

Reply:
Identity related Alert are very accurate in IDP, and in the M365D has created the same detection for Azure with that reference and future correlation objectives I have created this detection only for IDP